spec/unit/provider/package/zypper_spec.rb

```diff
@@ -200,6 +211,16 @@ def shell_out_expectation!(*command, **options)
       )
       provider.upgrade_package(["emacs"], ["1.0"])
     end
+
+    it "should run zypper upgrade with source option" do
+      new_resource.source "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      allow(::File).to receive(:exist?).with("/tmp/wget_1.11.4-1ubuntu1_amd64.rpm").and_return(true)
+      shell_out_expectation!(
+        "zypper", "--non-interactive", "install", "--auto-agree-with-licenses", "--oldpackage", "/tmp/wget_1.11.4-1ubuntu1_amd64.rpm"
+      )
+      provider.install_package(["wget"], ["1.11.4-1ubuntu1_amd64"])
+    end
+
```

Comment:
Could there be some tests here to show what happens:
1. when the version is given but does not match the source package
2. when the source file is missing
3. version is given

Reply:
agreed

Reply:
@PrajaktaPurohit  I have added functional and unit test cases for missing source path. 
Ready for review.